### ISSUE TITLE
Want to specify files to include, even when parent folder is excluded

### ISSUE THREAD

**Issue by punycode** [restic/restic#1361]
## Feature request

**TL;DR**: Want to nit-pick files or folders to be included in the backup, that would otherwise be excluded by an exclude pattern for a parent folder.

Coming from `duplicity`, we did make use of the ability to include a particular folder (hereby named `/mnt/private/actually-not-private`) into the backup, while excluding the parent folder (`/mnt/private`). Currently this cannot be expressed in `restic` backup options (at least not without crawling the whole filesystem tree externally and providing a complete backup file list via `--files-from`).

It would be a great help to have options, to instruct `restic` to scan & include these additional patterns, if they would be excluded otherwise:

```
  -i, --include pattern                  include a pattern (can be specified multiple times)
      --include-file file                read include patterns from a file (can be specified multiple times)
```

## Do you have an idea how to solve the issue?

While I don't actually know how to resolve this in actual code (just basic Go knowledge, no `restic` code base knowledge), I would delay the decision to exclude a particular path from the backup, if (and only if) there is an include pattern, that could be matched under that path.

I know that this would make scanning for files to consider a lot more taxing and I/O intensive. It could be designed as an additional _slow path_, documenting that using that feature will significantly slow down your backup runs.

PS: I would really prefer to not request that feature, but as usual: filesystem reality and legacy can be harsh.

PPS: If this would impose too much burden on simplicity of the code base, please close this issue instantly. We can still live with current backup solutions until these legacy layouts are gone.

**Comment by fd0** [comment]
Hey, thanks for raising this issue again. This limitation is already known, and while we don't have any immediate plans to tackle it, it is tracked in #233. So I'm going to close this issue here, if you want to be notified of any eventual progress, please subscribe in #233. Thanks!
